### ISSUE TITLE
API exception when trying to connect

### ISSUE THREAD

**Issue by jkobject** [broadinstitute/dalmatian#26]
when trying todo any action on Terra, I get this error message, preventing me from running any pipeline.

Do you have any idea what to do here?
I did `gcloud auth application-default login` a bunch and even reinstalled firecloud to the latest version but nothing seems to work.

```
APIException: Unable to query samples: (401) : <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head><script src="https://jsagent.tcell.io/tcellagent.min.js" tcellappid="FCProd-EBOgM" tcellapikey="AQQBBAFLGLOxL7VE9IF9ESlLvCxD5Ykr_7xkQKq_rgn_P58IWjOhOzIh6p3aI4pTWaprlUw" tcellbaseurl="https://us.agent.tcell.insight.rapid7.com/api/v1"></script>
<title>401 Unauthorized</title>
</head><body>
<h1>Unauthorized</h1>
<p>This server could not verify that you
are authorized to access the document
requested.  Either you supplied the wrong
credentials (e.g., bad password), or your
browser doesn't understand how to supply
the credentials required.</p>
<hr>
<address>Apache Server at api.firecloud.org Port 443</address>
</body></html>
```

Thanks!

**Comment by agraubert** [comment]
Terra has been talking for a while about switching to their own OAuth system, instead of using Google tokens, and apparently they finally made the switch. In their own infinite wisdom, they of course, did not tell anybody. I'll loop @francois-a and @dheiman into this, as this is going to require fixes at both the dalmatian and fiss stack levels. In the meantime, however, it looks like we're totally locked out of the API

**Comment by jkobject** [comment]
right when I am doing my depmap omics release :o.
Tell me if there is anything I can do to help.

Best,

**Comment by agraubert** [comment]
It appears that Terra had not actually made the switch _yet_, but they did have a different backend issue with authentication. The issue is resolved for now, but be on the lookout for their actual transition to their own OAuth system in the future

**Comment by gulatide** [comment]
Hello, I'm facing the exact same issue as OP. What was the solution for successfully making API calls to Terra, if any? 

For context, I'm using a VM to do this. So far, I've tried registering the service account with Terra by following Step 2 in [this guide](https://support.terra.bio/hc/en-us/articles/7448594459931-How-to-use-a-service-account-in-Terra). However, I still run into the same error as OP. I've also tried using my personal gcloud account, which works if I'm creating a job submission on my local machine, but not when I do it on the VM and I get the same error message.

Any insights would be appreciated. Thanks!
